buildSrc/src/main/java/org/elasticsearch/gradle/test/RestIntegTestTask.java

```diff
@@ -0,0 +1,114 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.test;
+
+import org.elasticsearch.gradle.SystemPropertyCommandLineArgumentProvider;
+import org.elasticsearch.gradle.testclusters.ElasticsearchCluster;
+import org.elasticsearch.gradle.testclusters.RestTestRunnerTask;
+import org.gradle.api.Action;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.NamedDomainObjectContainer;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+
+import java.util.function.Supplier;
+
+public class RestIntegTestTask extends DefaultTask {
+
+    protected RestTestRunnerTask runner;
+    private static final String TESTS_REST_CLUSTER = "tests.rest.cluster";
+    private static final String TESTS_CLUSTER = "tests.cluster";
+    private static final String TESTS_CLUSTER_NAME = "tests.clustername";
+
+    public RestIntegTestTask() {
+        Project project = getProject();
+        String name = getName();
+        super.dependsOn(project);
+        runner = project.getTasks().create(name + "Runner", RestTestRunnerTask.class);
+        NamedDomainObjectContainer<ElasticsearchCluster> testClusters = (NamedDomainObjectContainer<ElasticsearchCluster>) project
+            .getExtensions()
+            .getByName("testClusters");
+        ElasticsearchCluster cluster = testClusters.create(name);
+        runner.useCluster(cluster);
+        runner.include("**/*IT.class");
+        runner.systemProperty("tests.rest.load_packaged", Boolean.FALSE.toString());
+        if (System.getProperty(TESTS_REST_CLUSTER) == null) {
+            if (System.getProperty(TESTS_CLUSTER) != null || System.getProperty(TESTS_CLUSTER_NAME) != null) {
+                throw new IllegalArgumentException(
+                    String.format("%s, %s, and %s must all be null or non-null", TESTS_REST_CLUSTER, TESTS_CLUSTER, TESTS_CLUSTER_NAME)
+                );
+            }
+            SystemPropertyCommandLineArgumentProvider runnerNonInputProperties = (SystemPropertyCommandLineArgumentProvider) runner
+                .getExtensions()
+                .getByName("nonInputProperties");
+            runnerNonInputProperties.systemProperty(
+                TESTS_REST_CLUSTER,
+                (Supplier<String>) () -> String.join(",", cluster.getAllHttpSocketURI())
```

Comment:
Let's provide an overloaded method that takes a `Supplier<String>` so we don't have to do this explicit cast here.